internal/manifest/custom_resource.go

```diff
@@ -85,10 +85,6 @@ func PreDeleteDeleteCR(
 		return ErrWaitingForAsyncCustomResourceDeletion
```

Comment:
This function needs to be refactored further, the original description in the issue is not correct.

The block you need to remove is not `if !util.IsNotFound(err)` but this `if err == nil`

This function should: 
1. delete cr
2. if cr not found, further delete related CRD.
3. if CRD not found, remove manifest CR finalizer

So we can remove this wired `ErrWaitingForAsyncCustomResourceDeletion` when `err == nil`, as well as `ErrWaitingForAsyncCustomResourceDefinitionDeletion`

With the current change, basically this `PreDeleteDeleteCR` still block diff resource from being deleted. But due to you haven't covered this scenario in the test, it's not detected.

What you missing test is basically a module upgrade test when module CR is disabled, say we have v1.0.0 module contains resources A, B, C, and in upgrade v1.0.1 contains resources A, B, D. in this case, C will be deleted in this [step](https://github.com/kyma-project/lifecycle-manager/blob/c16f84e586922238616079ae78b58fc034393a01/internal/declarative/v2/reconciler.go#L482), but due to this module CR always return `ErrWaitingForAsyncCustomResourceDeletion`, it will prevent that upgrade happens.

So the test case should be designed like that
1. After module disabled.
2. apply a new moduletemplate with new version
3. verify all the resources in newer version get deployed.

For this test, we can use the moduletemplate within smoke test for module upgrade test [here](https://github.com/kyma-project/lifecycle-manager/blob/main/.github/workflows/test-smoke.yml#L150-L151)

Reply:
I will resolve the missing testcase in a follow-up PR